File: languages/php/Bloock/Entity/IdentityV2/DidParams.php
<?php

namespace Bloock\Entity\IdentityV2;

/**
 * Represents parameters used for generating DIDs.
 */
class DidParams
{
    private string $method;
    private string $blockchain;
    private string $network;

    /**
     * Constructs a DidParams object with the specified parameters.
     * @param string|null $method
     * @param string|null $blockchain
     * @param string|null $network
     */
    public function __construct(string $method = null, string $blockchain = null, string $network = null)
    {
        $this->method = $method;
        $this->blockchain = $blockchain;
        $this->network = $network;
    }

    public function toProto(): \Bloock\DidParams
    {
        $s = new \Bloock\DidParams();
        $s->setMethod(Method::toProto($this->method));
        $s->setBlockchain(Blockchain::toProto($this->blockchain));
        $s->setNetworkId(Network::toProto($this->network));

        return $s;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/DidParams.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

/**
 * Represents parameters used for generating DIDs.
 */
public class DidParams {
  private final Method method;
  private final Blockchain blockchain;
  private final Network network;

  /**
   * Construct a new instance of DidParams with default values.
   */
  public DidParams() {
    this.method = Method.POLYGONID;
    this.blockchain = Blockchain.POLYGON;
    this.network = Network.MUMBAI;
  }

  /**
   * Constructs a DidParams object with the specified parameters.
   * @param method
   * @param blockchain
   * @param network
   */
  public DidParams(Method method, Blockchain blockchain, Network network) {
    this.method = method;
    this.blockchain = blockchain;
    this.network = network;
  }

  public IdentityEntitiesV2.DidParams toProto() {
    return IdentityEntitiesV2.DidParams.newBuilder()
        .setMethod(this.method.toProto())
        .setBlockchain(this.blockchain.toProto())
        .setNetworkId(this.network.toProto())
        .build();
  }
}
